lib/meilisearch/rails/configuration.rb

```diff
@@ -11,7 +11,39 @@ def configuration=(configuration)
         @_config = configuration
       end
 
+      def deactivate!
+        semaphore.synchronize do
+          @_config.merge!(active: false)
+
+          return unless block_given?
+
+          yield
+
+          @_config.merge!(active: true)
+        end
+      end
+
+      def activate!
+        semaphore.synchronize do
+          @_config.merge!(active: true)
+        end
+      end
+
+      def active?
+        configuration.fetch(:active, true)
+      end
+
+      def black_hole
```

Comment:
What does this method do?

Reply:
I put some context [here](https://github.com/meilisearch/meilisearch-rails/issues/140#issuecomment-1193453248) but to improve my explanation...

The idea here is to have a way to keep calling the methods without receiving a `NoMethodError` because I wasn't able to wrap all the meilisearch calls with an `if` condition without breaking other parts of the code.

So now, any method I call when Meilisearch is disabled will result in `nil` or a `[]` when the method called is a `map` without error.

This `black_hole` is just a memoized instance of the singleton `NullObject`. Which will do what I explained earlier ;)

I'm not sure if it was veeery clear, but I recommend reading more about this design pattern called https://refactoring.guru/introduce-null-object